unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Catalog/SceneObjectCatalogController.cs

```diff
@@ -0,0 +1,389 @@
+
+using DCL.Helpers;
+using DCL.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.Networking;
+using UnityEngine.UI;
+
+public class SceneObjectCatalogController : MonoBehaviour 
+{
+    public System.Action<string> OnResultReceived;
+    public System.Action<SceneObject> OnSceneObjectSelected;
+    public System.Action<SceneObject, CatalogItemAdapter> OnSceneObjectFavorite;
+    public System.Action OnStopInput, OnResumeInput;
+
+    public Canvas generalCanvas;
+    public TextMeshProUGUI catalogTitleTxt;
+    public CatalogAssetPackListView catalogAssetPackListView;
+    public CatalogGroupListView catalogGroupListView;
+    public TMP_InputField searchInputField;
+    public RawImage[] shortcutsImgs;
+
+    List<Dictionary<string, List<SceneObject>>> filterObjects = new List<Dictionary<string, List<SceneObject>>>();
+    string lastFilterName = "";
+    bool catalogInitializaed = false, isShowingAssetPacks = false, isFavoriteFilterActive = false;
+    List<SceneObject> favoritesSceneObjects = new List<SceneObject>();
+    List<SceneObject> quickBarShortcutsSceneObjects = new List<SceneObject>() { null, null, null,null,null,null,null,null,null };
```

Comment:
This sequence of nulls in the variable definition don't seem very easy to maintain. Wouldn't be better to create the variable as an empty List and then initialize the list (for example in the awake or start method) as you want?

Reply:
This variable will be better described as an array. I use the list just for the methods of `Contains` and `IndexOf`, however, I can just swap to an array if you want like so `SceneObject[] quickBarShortcutsSceneObjects = new SceneObject[9];`

Reply:
Oh, ok! I didn't realize that it is being used as an array 👍 Ok then! Thanks Adrián!